docs/RELEASING.md

```diff
@@ -16,24 +11,26 @@ to be included with the Enterprise Search distribution.
 
 For example, when shipping for `8.1.2`, the version is `8.1.2.0`.
 
-To release Connectors:
+To release Connectors into Cloud Enterprise Search image:
 
 - Set the VERSION file to the new/incremented version on the release branch
 - Make sure all tests and linter pass with `make lint test`
 - PR these changes to the appropriate Connectors release branch
-- Run `make release`
```

Comment:
I just tried to explain the current process of releasing because it's different from what is written in the doc.

On the other hand I'm also not sure if it's correct, and as you've noticed I've missed the tagging part.

@tarekziade can you help me document the release steps in the doc?

Reply:
Ah, looks good!

I'll then close this PR and add link to the google doc in our PR template - cause PR template says "release ruby gems and try them in ent-search", but does not give an instruction how to do so.

I tried to approach it in this PR, but seems like it's just better for now link the doc.